File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
            int[] answer = new int[queries.length];

        for (int i = 0; i < queries.length; i++) {
            int min = Integer.MAX_VALUE;

            for (int j = queries[i][0]; j <= queries[i][1]; j++) {
                if (arr[j] > queries[i][2] && arr[j] < min) {
                    min = arr[j];
                }
            }

            answer[i] = (min == Integer.MAX_VALUE) ? -1 : min;
        }

        return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    let answer = [];
    for (let i = 0; i < queries.length; i++) {
        let min = Infinity;
        for (let j = queries[i][0]; j <= queries[i][1]; j++) {
            if (arr[j] > queries[i][2] && arr[j] < min) {
                min = arr[j];
            }
        }
        answer[i] = min === Infinity ? -1 : min;
    }
    return answer;
}